browser/ui/views/toolbar/brave_toolbar_view.cc

```diff
@@ -182,14 +206,25 @@ void BraveToolbarView::Update(content::WebContents* tab) {
     bookmark_->SetVisible(browser_defaults::bookmarks_enabled &&
                           edit_bookmarks_enabled_.GetValue());
   }
+  if (speedreader_) {
+    speedreader_->SetVisible(true);
+
+    auto* active_contents = GetWebContents();
+    if (active_contents) {
+      auto* tab_helper =
+          speedreader::SpeedreaderTabHelper::FromWebContents(active_contents);
+      if (tab_helper) {
+        speedreader_->SetActive(tab_helper->IsActiveForMainFrame());
+      }
+    }
+  }
```

Comment:
I think this logic could be moved into `SpeedreaderButton::Update(content::WebContents* web_contents)`.

Reply:
done